### PR TITLE
Migration histogram to use fastutil's primitive sorted map.

### PR DESCRIPTION
**Pull request by vjkoskela**
mad-treemap-to-fastutil-histogram

Green = New Version
Red = Existing Version (after all updates in this PR except Fastutil tree implementation change)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,38 +146,156 @@ Each of the pipeline configuration files should be placed in the *pipelinesDirec
...
+    "^memcached/memcached_ops/([^/]+)(/value)?": ["memcached/operations/$1"],
+    "^memcached/percent/([^/]+)(/value)?": ["memcached/$1"],
+    "^memcached/ps_count/.*": [],
+    "^memcached/ps_cputime/.*": [],
+    "^uptime/uptime(/value)?": ["uptime/value"]
+  },
+  "source": {
+    type="com.arpnetworking.metrics.common.sources.CollectdHttpSourceV1"
+    actorName="collectd-http-source"
+    name="telegraftcp_source"
```
Comment (BrandonArp) [inline README.md]: Misnamed here

File: config/pipelines/pipeline.conf [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  {
+    type="com.arpnetworking.tsdcore.sinks.PeriodFilteringSink"
+    name="telemetry_period_filtering_sink"
+    excludeGreaterThan="PT1S"
+    sink={
+      type="com.arpnetworking.tsdcore.sinks.TelemetrySink"
+      name="telemetry_sink"
+    }
+  }
+]
```
Comment (BrandonArp) [inline config/pipelines/pipeline.conf]: Newline

**Comment by vjkoskela** [comment]
mad-treemap-to-fastutil-histogram-2

Green = New Version
Red = Existing Version (after all updates in this PR except Fastutil tree implementation change)

In this test, both JVMs were given XMS=XMX=1024m. With no GC pressure (e.g. preallocated large heap) the new FastUtil implementation (green) grows much more slowly than the existing implementation (red). Consequently, it will collect garbage far more infrequently.

Therefore, if you set a high XMS you can reduce GC (and CPU) at the cost of memory consumption. If you set a low XMS you will increase GC (and CPU) but save memory. The same Telegraf workload used about 250Mb in the existing implementation and only about 50Mb in the new implementation.

A few more improvements like this and it's money.

**Comment by vjkoskela** [comment]
Most of the independent stuff was pulled into #86.

File: README.md (hunk 1 above)
Comment (vjkoskela) [reply README.md]: Fixed.

File: config/pipelines/pipeline.conf (hunk 2 above)
Comment (vjkoskela) [reply config/pipelines/pipeline.conf]: Fixed.

**Pull request merged**